WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Sniff to check theme for deregister and register of core scripts (jquery)
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_NoDeregisterCoreScriptSniff extends WordPress_Sniffs_Functions_FunctionRestrictionsSniff {
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_CONSTANT_ENCAPSED_STRING,
+			T_DOUBLE_QUOTED_STRING,
+			T_OPEN_PARENTHESIS,
+			T_CLOSE_PARENTHESIS,
+		);
+	}//end register()
+
+	/**
+	 * Process a given string and remove quotes.
+	 *
+	 * @param string $string The text to be processed.
+	 *
+	 * @return string
+	 */
+	function trimquotes( $string ) {
+		return trim( $string, '"\'' );
+	}
+
+	/**
+	 * An array of core scripts to check for.
+	 *
+	 * @return array
+	 */
+	function corescripts() {
```

Comment:
As this function doesn't *do* anything, the array should be a class property instead.